src/Main.elm

```diff
@@ -25,11 +46,49 @@ init =
 
 type Msg
     = NoOp
+    | EditSettings
+    | TaskSettingsMsg TaskSettings.Msg
+    | EditCanceled
+    | SettingsUpdated SettingsDto
 
 
 update : Msg -> Model -> ( Model, Cmd Msg )
-update _ model =
-    ( model, Cmd.none )
+update msg model =
+    case ( model.page, msg ) of
```

Comment:
It might be hard to navigate all the options here. Is there a way on your mind how to improve readability of the code here?